codegen/lib/graphql_java_gen.rb

```diff
@@ -274,4 +274,34 @@ def abstract_types
       end
     end
   end
+
+  def java_doc(element)
+    doc = ''
+    unless element.description.nil?
+      description = wrap_text(element.description, 100)
+      description = format_java_doc_list(description)
+      description = description.chomp("\n").gsub("\n", "\n* ")
+      doc << "/**\n"
+      doc << '* '
+      doc << description
+      doc << "\n*/"
+    end
+    doc
+  end
+
+  def wrap_text(text, col_width=80)
+    text.gsub!( /(\S{#{col_width}})(?=\S)/, '\1 ' )
+    text.gsub!( /(.{1,#{col_width}})(?:\s+|$)/, "\\1\n" )
+    text
+  end
+
+  def format_java_doc_list(text)
+    parts = text.split(':')
+    list_pattern = /\s*?-\s*?`(.+?)`\s*?/
```

Comment:
Why are you assuming lists have backticks in them?  That is to format text as code in markdown, which is unrelated to lists.  This seems very coupled to specific types of lists we have in Shopify's schema.